src/txkube/test/test_swagger.py

```diff
@@ -584,6 +590,32 @@ def test_array_property(self):
         )
 
 
+    def test_array_property_py3(self):
+        """
+        For a definition with a property of type *array*,
+        ``Swagger.pclass_for_definition`` returns a ``PClass`` with a
+        corresponding field which requires a ``PVector`` of some type.
+        """
+        if version_info < (3,):
+            self.skipTest("skipping test on Python 2")
+
+        spec = Swagger.from_path(self.spec_path)
+        # Arbitrarily select a simple definition that includes an array from
+        # the spec.  It also demonstrates that "required" itself is optional.
+        name = u"v1.Capabilities"
+        Capabilities = spec.pclass_for_definition(name)
+
+        # On Python 3, type(b"hello"[0]) is int
+        self.expectThat(
+            lambda: Capabilities(add=b"hello"),
+            raises_exception(
+                CheckedValueTypeError,
+                expected_types=(unicode,),
+                actual_type=int,
+            ),
+        )
```

Comment:
Instead of doubling this test, how about replacing the bad value with a less fraught type?  For example, `add=[0]` should fail roughly the same way as `b"hello"` does but without the varying behavior between Python 2 and Python 3.

Reply:
Sounds good to me.  I have updated the test to use `add=[0]`.